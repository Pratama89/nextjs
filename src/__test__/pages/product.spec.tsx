import { describe } from "node:test"
import { render, screen } from "@testing-library/react"
import ProductPage from "@/pages/product"

jest.mock("next/router", () => {
  return {
    useRouter() {
      return {
        route: "/product",
        pathname: "",
        query: "",
        asPath: "",
        push: jest.fn(),
        events: {
          on: jest.fn(),
          off: jest.fn(),
        },
        beforePopState: jest.fn(() => null),
        prefetch: jest.fn(() => null),
        isReady: true,
      }
    }
  }
})

describe("Product Page", () => {
    it("should render the Product Page", () => {
        const page = render(<ProductPage />)
        // expect(screen.getByTestId("title").textContent).toBe("Product Pages")
        expect(page).toMatchSnapshot()
    })
})