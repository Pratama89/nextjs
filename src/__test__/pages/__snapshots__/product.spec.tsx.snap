// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Product Page 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class=""
      >
        <div
          class="w-full p-5 mb-5"
        >
          <h1
            class="text-2xl font-bold mt-5 text-center  "
          >
            Product
          </h1>
          <div
            class="flex flex-wrap mx-auto lg:px-6 px-2 justify-evenly lg:justify-start md:justify-start mt-5 mb-5 border bg-slate-100 rounded-lg overflow-hidden"
          >
            <div
              class="w-1/4 p-3 animate-pulse"
            >
              <div
                class="w-full aspect-square rounded-sm bg-slate-400"
              />
              <div
                class="w-full h-5 mt-1 bg-slate-400 rounded-sm"
              />
              <div
                class="w-full h-4 mt-1 bg-slate-400 rounded-sm"
              />
              <div
                class="w-full h-4 mt-2 bg-slate-400 rounded-sm"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class=""
    >
      <div
        class="w-full p-5 mb-5"
      >
        <h1
          class="text-2xl font-bold mt-5 text-center  "
        >
          Product
        </h1>
        <div
          class="flex flex-wrap mx-auto lg:px-6 px-2 justify-evenly lg:justify-start md:justify-start mt-5 mb-5 border bg-slate-100 rounded-lg overflow-hidden"
        >
          <div
            class="w-1/4 p-3 animate-pulse"
          >
            <div
              class="w-full aspect-square rounded-sm bg-slate-400"
            />
            <div
              class="w-full h-5 mt-1 bg-slate-400 rounded-sm"
            />
            <div
              class="w-full h-4 mt-1 bg-slate-400 rounded-sm"
            />
            <div
              class="w-full h-4 mt-2 bg-slate-400 rounded-sm"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
